src/main/java/org/opensearch/plugin/insights/rules/model/healthStats/QueryGrouperHealthStats.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.rules.model.healthStats;
+
+import java.io.IOException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+/**
+ * Represents the health statistics of the query grouper.
+ */
+public class QueryGrouperHealthStats implements ToXContentFragment, Writeable {
+    private final int queryGroupCount;
+    private final int queryGroupHeapSize;
+    private static final String QUERY_GROUP_COUNT = "QueryGroupCount";
+    private static final String QUERY_GROUP_HEAP_SIZE = "QueryGroupHeapSize";
```

Comment:
I originally thought this measured memory usage. Wondering if we should rename to "count" to avoid confusion even though we have multiple underlying data structures (priority queue, max heap).

Reply:
Updated!